battle-engine.js

```diff
@@ -510,6 +506,16 @@ BattlePokemon = (function () {
 		}
 		return null;
 	};
+	BattlePokemon.prototype.isSuppressing = function (effectType) {
```

Comment:
I chose this method because it allowed me to run off with a mass replace using something like `/\.ignore\[([^\]])\]/.isSuppressing($1)/g` and fix 95% of functionality before manually replacing any outliers.